src/nodes.coffee

```diff
@@ -1104,47 +1115,77 @@ exports.Obj = class Obj extends Base
 
   children: ['properties']
 
+  isAssignable: ->
+    @lhs = true
```

Comment:
I think it's pretty close... I'd like to run it against a codebase or two and maybe add a few tests.

Would you prefer it if desrtuctured parameters went in this PR, or would you be happy for them to be tackled separately?